### PR TITLE
Add JSON renderer

### LINKED ISSUE #10: Add switch for rendering JSON output
It would be great to have a JSON renderer so you can store or process the output. Instead of a table the output could be something like
```json
[
  {
    "name": "blessed",
    "packageVersion": "0.1.81",
    "installedVersion": "0.1.81",
    "matches": true,
    "locked": false
  },
]
```

### PR DESCRIPTION
**Pull request by timjacobi**
resolves #10

### CODE REVIEW

File: src/renderers/json.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+{
+  /*
+   * Merge dependencies and dev dependencies
+   */
+  const allPackages = depMap.deps.concat(depMap.devDeps);
+
+  /*
+   * Print the JSON
+   */
+  console.log(JSON.stringify(createMatchInfos(allPackages), undefined, 2));
```
Comment (andrewhathaway) [inline src/renderers/json.ts]: Can we split this line up a little bit? Just to increase some readability!

**Comment by andrewhathaway** [comment]
Hi @timjacobi!

This look great, 🔥 !  I've left a comment, it'd be great if you could see to that. Also, please update the README.md information with the renderer option if possible!

Looking great.

**Comment by timjacobi** [comment]
No worries. Just updated as requested :)

File: README.md [hunk 2]
```diff
@@ -53,6 +53,37 @@ The `blessed` renderer will render a table inside a screen, that has be exited b
 
 ![](./.github/blessed.png?raw=true)
 
+### JSON
+The `JSON` renderer will print a JSON array with a match info for each package
```
Comment (andrewhathaway) [inline README.md]: Can we get a full stop here pls!

**Comment by andrewhathaway** [comment]
@timjacobi Last comment, then I'm happy to merge!

File: README.md (hunk 2 above)
Comment (juliankay) [reply README.md]: Not sure about the wording on this... does the term 'a match info' make sense here? Can we just have something like 'the match information'?

**Review by andrewhathaway — APPROVED** [review]
Looking great!

**Comment by andrewhathaway** [comment]
Thank you @timjacobi!

**Pull request merged**